### PR TITLE
Added random split for non-SMILES identifiers

### PR DESCRIPTION
**Pull request by AdrianM0**

### CODE REVIEW

File: data/tabular/train_test_split.py [hunk 1]
```diff
@@ -197,40 +214,86 @@ def cli(
...
-    #     elif "freesolv" in path:
+    #     elif "BBBP" in path:
     #         filtered_paths.append(path)
     #     elif "peptide" in path:
     #         filtered_paths.append(path)
+    #     elif "bicerano_dataset" in path:
+    #         filtered_paths.append(path)
     # paths_to_data = filtered_paths
 
     REPRESENTATION_LIST = []
```
[the remark this answers is not in the document]
Comment (AdrianM0) [reply data/tabular/train_test_split.py:225]: yes, that is a global variable in LN33, but I think it is a good idea to change this local one to proper convention with lower-case names 🤔

File: data/tabular/train_test_split.py [hunk 2]
```diff
@@ -197,40 +214,86 @@ def cli(
...
-        if repr_col in df.columns:
+
+        if repr_col in df.columns and path not in TRACKED_DATASETS:
+            print("Processing", path.split("/")[0])
+            TRACKED_DATASETS.append(path)
             REPRESENTATION_LIST.extend(df[repr_col].to_list())
 
     REPR_DF = pd.DataFrame()
-    REPR_DF["SMILES"] = list(set(REPRESENTATION_LIST))
+    REPR_DF[repr_col] = list(set(REPRESENTATION_LIST))
```
Comment (kjappelbaum) [inline data/tabular/train_test_split.py:236]: in general, there should not need to be a need to allocate a dataframe for this (which can cost more memory). But it should work for our needs.

File: data/tabular/train_test_split.py [hunk 3]
```diff
@@ -197,40 +214,86 @@ def cli(
...
-        paths_to_data, merge, override=override, check=check, repr_col=repr_col
+        csv_paths=paths_to_data,
+        repr_col=repr_col,
+        train_test_df=merge,
+        override=override,
+        check=check,
     )
 
 
+def cli(
```
Comment (kjappelbaum) [inline data/tabular/train_test_split.py:279]: nice, this should do what we want it to do. I'll walk through it once more tomorrow with a fresh mind

File: data/tabular/train_test_split.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (AdrianM0) [reply data/tabular/train_test_split.py]: @MicPie now should be better

File: data/tabular/scaffold_split.py [hunk 4]
```diff
@@ -0,0 +1,182 @@
...
+                    cwd=os.path.dirname(file),
+                )
+
+            transformed_files.append(
+                os.path.join(os.path.dirname(file), "data_clean.csv")
+            )
+
+    all_smiles = set()
+    for file in transformed_files:
+        df = pd.read_csv(file)
```
Comment (MicPie) [inline data/tabular/scaffold_split.py:130]: add `low_memory=False`?

File: data/tabular/scaffold_split.py [hunk 5]
```diff
@@ -0,0 +1,182 @@
...
+        with open(file, "r") as f:
+            meta = yaml.safe_load(f)
+
+        if meta["name"] in to_scaffold_split:
+            # run transform.py script in this directory if `data_clean.csv` does not exist
+            # use this dir as cwd
+            if not os.path.exists(
+                os.path.join(os.path.dirname(file), "data_clean.csv")
+            ):
+                subprocess.run(
```
Comment (MicPie) [inline data/tabular/scaffold_split.py:119]: When you are running "transform.py" you are not ensuring that the additional mol. reprs. are present. But we can keep it as it is, but just to mention that here.

File: data/tabular/scaffold_split.py [hunk 6]
```diff
@@ -0,0 +1,182 @@
...
+    # select the right indices for each split
+    train_smiles = [all_smiles[i] for i in splits["train"]]
+    val_smiles = [all_smiles[i] for i in splits["valid"]]
+    print(
+        "Train smiles:",
+        len(train_smiles),
+        "Val smiles:",
+        len(val_smiles),
+        "Test smiles:",
+        len(splits["test"]),
```
Comment (MicPie) [inline data/tabular/scaffold_split.py:148]: why no `test_smiles` here and above?

File: src/chemnlp/data/split.py
Comment (MicPie) [inline src/chemnlp/data/split.py:1]: No changes here as only moved, right?

**Review by MicPie — APPROVED** [review]

File: src/chemnlp/data/split.py
Comment (kjappelbaum) [reply src/chemnlp/data/split.py:1]: hm, there is new content. Let me ensure i pushed latest version

File: data/tabular/scaffold_split.py (hunk 6 above)
Comment (kjappelbaum) [reply data/tabular/scaffold_split.py:148]: cause we do not allocate a list of smiles for test (we only have the indices in the dict)

File: data/tabular/scaffold_split.py (hunk 4 above)
Comment (kjappelbaum) [reply data/tabular/scaffold_split.py:130]: good point

**Pull request merged**